s3/s3.go

```diff
@@ -55,6 +66,7 @@ type S3ClientOpts struct {
 	RoleARN         string
 	RoleSessionName string
 	UseSDKCreds     bool
+	EncryptOpts     *EncryptOpts
```

Comment:
minor - if this was non-pointer - you could avoid nil checks

Reply:
Should it be optional? I don't have a good grasp on how this project is used by the other argo projects. If its non-optional, that would require downstream projects to fix their code when they update the library right? Would I need to update all the downstream projects?

Reply:
If it were non pointer, then enabled would be the zero value, false.

Reply:
Updated